src/registrar/utility/csv_export.py

```diff
@@ -32,893 +33,1132 @@ def write_header(writer, columns):
-    dict_user_domain_roles,
-):
-    """Helper function to update columns with domain manager information"""
-
-    domain_name = domain_info.domain.name
-
-    try:
-        dms_active, dms_invited = count_domain_managers(
-            domain_name, dict_domain_invitations_with_invited_status, dict_user_domain_roles
+
+    @classmethod
+    def model(cls):
+        # Return the model class that this export handles
+        return DomainInformation
+
+    @classmethod
+    def update_queryset(cls, queryset, **kwargs):
+        """
+        Returns an updated queryset.
+
+        Add security_contact_email, invited_users, and managers to the queryset,
+        based on public_contacts, domain_invitations and user_domain_roles
+        passed through kwargs.
+        """
+        public_contacts = kwargs.get("public_contacts", {})
+        domain_invitations = kwargs.get("domain_invitations", {})
+        user_domain_roles = kwargs.get("user_domain_roles", {})
+
+        annotated_domain_infos = []
+
+        # Create mapping of domain to a list of invited users and managers
+        invited_users_dict = defaultdict(list)
+        for domain, email in domain_invitations:
+            invited_users_dict[domain].append(email)
+
+        managers_dict = defaultdict(list)
+        for domain, email in user_domain_roles:
+            managers_dict[domain].append(email)
+
+        # Annotate with security_contact from public_contacts
+        for domain_info in queryset:
+            domain_info["security_contact_email"] = public_contacts.get(
+                domain_info.get("domain__security_contact_registry_id")
+            )
+            domain_info["invited_users"] = ", ".join(invited_users_dict.get(domain_info.get("domain__name"), []))
+            domain_info["managers"] = ", ".join(managers_dict.get(domain_info.get("domain__name"), []))
+            annotated_domain_infos.append(domain_info)
+
+        if annotated_domain_infos:
+            return annotated_domain_infos
+
+        return queryset
+
+    # ============================================================= #
+    # Helper functions for django ORM queries.                      #
+    # We are using these rather than pure python for speed reasons. #
+    # ============================================================= #
+
+    @classmethod
+    def get_all_security_emails(cls):
+        """
+        Fetch all PublicContact entries and return a mapping of registry_id to email.
+        """
+        public_contacts = PublicContact.objects.values_list("registry_id", "email")
+        return {registry_id: email for registry_id, email in public_contacts}
+
+    @classmethod
+    def get_all_domain_invitations(cls):
+        """
+        Fetch all DomainInvitation entries and return a mapping of domain to email.
+        """
+        domain_invitations = DomainInvitation.objects.filter(status="invited").values_list("domain__name", "email")
+        return list(domain_invitations)
```

Comment:
good question.  i normalized these to all return dicts.

Reply:
Okay, had to revert this.  They are different because there can be a list of domain invitations and a list of user domain roles per domain, where there is one security email per domain.